cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceRuntimeService.java

```diff
@@ -257,11 +260,25 @@ protected void startUp() throws Exception {
           classpath.add(logbackLocation.getName());
         }
 
-        // Generate and localize the launcher jar to control the classloader of MapReduce containers processes
-        classpath.add("job.jar/lib/*");
+        // Get all the jars in jobJar and sort them lexically before adding to the classpath
+        // This allows CDAP classes to be picked up first before the Twill classes
+        JarFile jobJarFile = new JarFile(jobJar);
+        Enumeration<JarEntry> entries = jobJarFile.entries();
+        List<String> jarFiles = new ArrayList<>();
+        while (entries.hasMoreElements()) {
+          JarEntry entry = entries.nextElement();
+          if (entry.getName().endsWith(".jar")) {
```

Comment:
This is not the same as the existing logic. We should only include jars under the lib dir